kroxylicious-operator/pom.xml

```diff
@@ -159,6 +159,12 @@
             <artifactId>log4j-slf4j2-impl</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>io.strimzi</groupId>
```

Comment:
The additional dependency gave me pause for thought.  Do we want the strimzi-api as a runtime dependency? Does this set a precedent for how we'll integrate other kafka providers (confluent, redpanda etc)?  I worry about ballooning dependencies and what might happen as providers evolve their Kube APIs.

The alternative **might** be https://github.com/operator-framework/java-operator-sdk/issues/2126 and https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md#resource-typeless-api.  There are disadvantages with the typeless API... we lose type safety and issues that would be a compile time become runtime ones.

I think I'm content with the using the strimzi-api, at least for now.  If it turns out to be problematic later, we can refactor.

Reply:
For the record, I'm comfortable with a fabric8 style API dependency jar for Strimzi and any other provider on the classpath but would be far less comfortable with any heavier weight depdendencies.